examples/tic-tac-toe/move/sources/owned.move

```diff
@@ -0,0 +1,326 @@
+        id.delete();
+
+        // Confirm that the mark is from the player we expect -- it should not
+        // be possible to hit this assertion, because the `Mark`s can only be
+        // created by the address that owns the `TurnCap` which cannot be
+        // transferred, and is always held by `game.next_player()`.
+        let (me, them, sentinel) = game.next_player();
+        assert!(me == player, EWrongPlayer);
+
+        if (game[row, col] == MARK__) {
+            *(&mut game[row, col]) = sentinel;
+            game.turn = game.turn + 1;
+        };
+
+        // Check win condition -- if there is a winner, send them the trophy,
+        // otherwise, create a new turn cap and send that to the next player.
+        let end = game.ended();
+        if (end == TROPHY_WIN) {
+            transfer::transfer(game.mint_trophy(end, them, ctx), me);
+            event::emit(GameEnd { game: object::id(game) });
+        } else if (end == TROPHY_DRAW) {
+            transfer::transfer(game.mint_trophy(end, them, ctx), me);
+            transfer::transfer(game.mint_trophy(end, me, ctx), them);
+            event::emit(GameEnd { game: object::id(game) });
+        } else if (end == TROPHY_NONE) {
+            let cap = TurnCap { id: object::new(ctx), game: object::id(game) };
+            let (to, _, _) = game.next_player();
+            transfer::transfer(cap, to);
+        } else {
+            abort EInvalidEndState
+        }
+    }
+
+    public fun burn(game: Game) {
+        assert!(game.ended() != TROPHY_NONE, ENotFinished);
+        let Game { id, .. } = game;
+        id.delete();
+    }
+
+    /// Test whether the game has reached an end condition or not.
+    public fun ended(game: &Game): u8 {
+        if (
+            // Test rows
+            test_triple(game, 0, 1, 2) ||
+            test_triple(game, 3, 4, 5) ||
+            test_triple(game, 6, 7, 8) ||
+            // Test columns
+            test_triple(game, 0, 3, 6) ||
+            test_triple(game, 1, 4, 7) ||
+            test_triple(game, 2, 5, 8) ||
+            // Test diagonals
+            test_triple(game, 0, 4, 8) ||
+            test_triple(game, 2, 4, 6)
+        ) {
+            TROPHY_WIN
+        } else if (game.turn == 9) {
+            TROPHY_DRAW
+        } else {
+            TROPHY_NONE
+        }
+    }
+
+    #[syntax(index)]
+    public fun mark(game: &Game, row: u8, col: u8): &u8 {
+        &game.board[(row * 3 + col) as u64]
+    }
+
+    #[syntax(index)]
+    fun mark_mut(game: &mut Game, row: u8, col: u8): &mut u8 {
+        &mut game.board[(row * 3 + col) as u64]
+    }
+
+    // === Private Helpers ===
+
+    /// Address of the player the move is expected from, the address of the
+    /// other player, and the mark to use for the upcoming move.
+    fun next_player(game: &Game): (address, address, u8) {
+        if (game.turn % 2 == 0) {
+            (game.x, game.o, MARK_X)
+        } else {
+            (game.o, game.x, MARK_O)
+        }
```

Comment:
```suggestion
        if (game.turn % 2 == 0) (game.x, game.o, MARK_X)
        else (game.o, game.x, MARK_O)
```

Reply:
Hmm, I'm not a huge fan of this change, because not having the two cases aligned makes it difficult to note the differences between them.

Reply:
```suggestion
        if (game.turn % 2 == 0) (game.x, game.o, MARK_X)
        else                    (game.o, game.x, MARK_O)
```

Reply:
Kidding of course :p
And fair enough. I would always say to do what you think is more readable

Reply:
Don't tempt me, in a past life, I've been [known to do worse](https://github.com/amnn/cells/blob/fdb90afe7864c14c03ac58cb7e6940264e1da9c8/src/GLFWScr.cpp#L17-L22).